src/llama_stack/providers/inline/agents/meta_reference/responses/streaming.py

```diff
@@ -722,10 +722,12 @@ async def _process_streaming_chunks(
                                 )
 
                             # Accumulate arguments for final response (only for subsequent chunks)
-                            if not is_new_tool_call:
-                                response_tool_call.function.arguments = (
-                                    response_tool_call.function.arguments or ""
-                                ) + tool_call.function.arguments
+                            if not is_new_tool_call and response_tool_call is not None:
+                                # Need to check function is not None
+                                if response_tool_call.function and tool_call.function:
```

Comment:
this `if X_tool_call.function` checks are annoying. if the tool call exists, its function field is always non null or else we wouldn't come so far down right? i wonder if there is a way to either assert or write things slightly differently to make this easier to see (both for us and the type checker)